pysat/utils/io.py

```diff
@@ -0,0 +1,1159 @@
+        desired. Ignored if zlib=False. (default=4)
+    shuffle : bool
+        The HDF5 shuffle filter will be applied before compressing the data.
+        This significantly improves compression. Ignored if zlib=False.
+        (default=True)
+    preserve_meta_case : bool
+        Flag specifying the case of the meta data variable strings. If True,
+        then the variable strings within the MetaData object (which
+        preserves case) are used to name variables in the written netCDF
+        file. If False, then the variable strings used to access data from
+        the pysat.Instrument object are used instead. (default=False)
+    export_nan : list or NoneType
+        By default, the metadata variables where a value of NaN is allowed
+        and written to the netCDF4 file is maintained by the Meta object
+        attached to the pysat.Instrument object. A list supplied here
+        will override the settings provided by Meta, and all parameters
+        included will be written to the file. If not listed
+        and a value is NaN then that attribute simply won't be included in
+        the netCDF4 file. (default=None)
+    unlimited_time : bool
+        Flag specifying whether or not the epoch/time dimension should be
+        unlimited; it is when the flag is True. (default=True)
+
+    Note
+    ----
+    Stores 1-D data along dimension 'epoch' - the date time index.
+
+    Stores higher order data (e.g. dataframes within series) separately
+
+    - The name of the main variable column is used to prepend subvariable
+      names within netCDF, var_subvar_sub
+    - A netCDF4 dimension is created for each main variable column
+      with higher order data; first dimension Epoch
+    - The index organizing the data stored as a dimension variable
+    - from_netcdf4 uses the variable dimensions to reconstruct data
+      structure
+
+    All attributes attached to instrument meta are written to netCDF attrs
+    with the exception of 'Date_End', 'Date_Start', 'File', 'File_Date',
+    'Generation_Date', and 'Logical_File_ID'. These are defined within
+    to_netCDF at the time the file is written, as per the adopted standard,
+    SPDF ISTP/IACG Modified for NetCDF. Atrributes 'Conventions' and
+    'Text_Supplement' are given default values if not present.
+
+    """
+    # Check export NaNs first
+    if export_nan is None:
+        export_nan = inst.meta._export_nan
+
+    # Ensure the metadata is set and updated to netCDF4 standards
+    add_netcdf4_standards_to_meta(inst, epoch_name)
+
+    # Base_instrument used to define the standard attributes attached
+    # to the instrument object. Any additional attributes added
+    # to the main input Instrument will be written to the netCDF4
+    if base_instrument is None:
+        base_attrb = dir(pysat.Instrument())
+    else:
+        warnings.warn("".join(["`base_instrument` has been deprecated and will",
```

Comment:
Why is base_instrument being deprecated?

Reply:
I didn't see why it was useful.  This is one of the reasons this is a draft PR.

Reply:
TODO: add issue for new usage of base_instrument